### DISCUSSION TITLE
Create new tab / window via cli

### DISCUSSION THREAD

**Discussion by ricoberger** [ghostty-org/ghostty#3445] [Feature Requests, Ideas] [answered]
It would be great if it is possible to create new tabs / windows via the ghostty cli, e.g. by running `ghostty +new-tab -c <command>` 

Use case: I'm using [gh-dash](https://github.com/dlvhdr/gh-dash), with the following keybindings, to open an issue / pr in a new tmux window and automatically open it in nvim:

```
keybindings:
  issues:
    - key: O
      command: tmux new-window -n "{{.RepoName}}/{{.IssueNumber}}" -c {{.RepoPath}} 'nvim -c ":Octo issue edit {{.IssueNumber}}"'
  prs:
    - key: D
      command: tmux new-window -n "{{.RepoName}}/{{.PrNumber}}" -c {{.RepoPath}} 'gh pr checkout {{.PrNumber}} && nvim -c ":DiffviewOpen main...{{.HeadRefName}}"'
    - key: O
      command: tmux new-window -n "{{.RepoName}}/{{.PrNumber}}" -c {{.RepoPath}} 'nvim -c ":Octo pr edit {{.PrNumber}}"'
```

It would be great if the same can be achieved directly with ghostty, so that tmux is not required for such a use cases.

**Comment by JakeWharton** [comment, marked as answer]
Covered by #2353

**Reply by flypenguin** [reply]
that linked thread is insanely huge – a "TL;DR" oneliner would have been super helpful :/

UPDATE:

This seems (as of 2025-05-22) not possible, as indicated in [this comment](https://github.com/ghostty-org/ghostty/discussions/2353#discussioncomment-12617469). Shame :/

**Reply by 00-kat** [reply]
TL;DR: not currently possible, go subscribe to #2353.

**Comment by Elijas** [comment]
## Workaround (not a solution) until #2353 is implemented

You can open a new window (macOS) with
```bash
open -na ghostty --args --title=ghostty-from-vscode --working-directory="$(pwd)"
```

To make it also run a command (`echo hello`) on startup:
```bash
open -na ghostty --args --title=ghostty-from-vscode --working-directory="$(pwd)" -e sh -c 'echo hello; exec $SHELL'
```

**Reply by Elijas** [reply]
For completeness, there's also an official `+new-window` option that probably has the same effect
https://github.com/ghostty-org/ghostty/discussions/4579#discussioncomment-14644755
it doesn't work on macOS though

**Comment by keptt** [comment]
if someone needs to open a new tab rather than a new window in macOS, the following worked great for me: `open -a ghostty <path_to_the_dir_to_open>`. My usecase was to open the directory of the currently opened file in neovim

**Reply by Elijas** [reply]
is it possible to make it run a command upon opening?

For me this `open -a ghostty /bin/sh` kind of works (but mac asks to approve security warning though)

**Reply by keptt** [reply]
it would be nice to make it work from the command line itself, but the only way I know how to do it is to add the following:
```
command=<full_path_to_your_executable>
```
into the ghostty config file, i.e. `~/.config/ghostty/config`

**Reply by keptt** [reply]
another related thread: https://github.com/ghostty-org/ghostty/discussions/4254

**Reply by mrw1986** [reply]
@Elijas If you use fish you can make it work using a fish startup hook. For instance, Claude Code can create a new worktree, open a Ghostty tab, and also inject a Claude Code prompt automatically. Here's a repo I made with everything you need: https://github.com/mrw1986/ghostty-tab-launch

**Reply by Elijas** [reply]
> @Elijas If you use fish you can make it work using a fish startup hook. For instance, Claude Code can create a new worktree, open a Ghostty tab, and also inject a Claude Code prompt automatically. Here's a repo I made with everything you need: https://github.com/mrw1986/ghostty-tab-launch

Sweet! Thanks
